### ISSUE TITLE
read suggestions on training NNs

### ISSUE THREAD

**Issue by ismael-mendoza** [ismael-mendoza/bliss#63]
https://karpathy.github.io/2019/04/25/recipe/

**Comment by ismael-mendoza** [comment]
Takeaways: 

- know your data very very well, without any NN involved; make histograms, see imbalances, etc. 
- see section 2. for an excellent set of steps by step to debugging any network i.e. to get it going when it's not training anywhere close to optimal performance
- visualize prediction dynamics
- Adam with `3e-4` learning rate is always a good place to start
- start by complexifying the network and the data step-by-step. At each step verify you get performance boost you expect. 
- *decrease batch_size*: Because of BatchNorm structure, lower batch_size corresponds to stronger regularization
- adding more data to your training set is the best way to regularize
- dropout and batchnorm don't seem to play nice
- don't trust weight_decay defaults, set to 0 initially and tune at the very end
- *at the end once you are happy with performance*: (1) ensemble of models usually increase accuracy by 2% (~ 5 models) (2) leave it training for a LONG time
